src/main/java/net/snowflake/ingest/utils/ParameterProvider.java

```
@@ -411,4 +433,15 @@ public Constants.BdecParquetCompression getBdecParquetCompressionAlgorithm() {
   public String toString() {
     return "ParameterProvider{" + "parameterMap=" + parameterMap + '}';
   }
+
+  private void icebergModeValidation(String key, Object expected) {
+    Object val = this.parameterMap.get(key);
+    if (!val.equals(expected)) {
```

We should throw an exception instead of just a warning